tests/UKHO.S100PermitService.Common.UnitTests/Services/ManufacturerKeyServiceTests.cs

```diff
@@ -0,0 +1,127 @@
+﻿using Azure.Security.KeyVault.Secrets;
+using FakeItEasy;
+using FluentAssertions;
+using Microsoft.Extensions.Options;
+using UKHO.S100PermitService.Common.Clients;
+using UKHO.S100PermitService.Common.Configuration;
+using UKHO.S100PermitService.Common.Exceptions;
+using UKHO.S100PermitService.Common.Providers;
+using UKHO.S100PermitService.Common.Services;
+
+namespace UKHO.S100PermitService.Common.UnitTests.Services
+{
+    [TestFixture]
+    public class ManufacturerKeyServiceTests
+    {
+        private IOptions<ManufacturerKeyConfiguration> _fakeManufacturerKeyVault;
+        private ICacheProvider _fakeCacheProvider;
+        private ISecretClient _fakeSecretClient;
+        private IManufacturerKeyService _manufacturerKeyService;
+
+        [SetUp]
+        public void Setup()
+        {
+            _fakeManufacturerKeyVault = A.Fake<IOptions<ManufacturerKeyConfiguration>>();
+            _fakeCacheProvider = A.Fake<ICacheProvider>();
+            _fakeSecretClient = A.Fake<ISecretClient>();
+
+            _fakeManufacturerKeyVault.Value.CacheTimeoutInHours = 2;
+
+            _manufacturerKeyService = new ManufacturerKeyService(_fakeManufacturerKeyVault, _fakeCacheProvider, _fakeSecretClient);
```

Comment:
Can we also assert ArgumentNullException. Please refer other test files.

Reply:
Added